### PR TITLE
Remove dependency between chip-tool and the nlunit-test

### PR DESCRIPTION
**Pull request by jlatusek**
This change is related to #29682. 
## Problem
The chip-tool has in dependency to nlunit-test trought other targets:
``` sh
$ gn path out/linux-x64-chip-tool/ //third_party/nlunit-test:nlunit-test //examples/chip-tool
//examples/chip-tool:chip-tool --[private]-->
//examples/chip-tool:chip-tool-utils --[public]--> 
//src/app/tests/suites/commands/interaction_model:interaction_model --[public]-->
//src/lib/support:testing --[public]-->
//third_party/nlunit-test:nlunit-test
``` 
This causes a problem while replacing the nlunit-test with pigweed ones. 
Pigweed unit tests require setting the proper backend for used modules. Without it, the compilation ends up with an error. The build of the chip-tool should not set variables related to unit testing.

## Changes

I've split testing into two new targets: `testing_nlunit` and `test_utils`.  `testing_nlunit`  will be removed entirely after the end of the integration of pigweed tests. 

## Testing
Ci will test it, the dependency can be check by:
``` sh 
./scripts/build/build_examples.py --target linux-x64-chip-tool build
gn path out/linux-x64-chip-tool/ //third_party/nlunit-test:nlunit-test //examples/chip-tool
```

### CODE REVIEW

File: examples/chip-tool/BUILD.gn
Comment (bzbarsky-apple) [inline examples/chip-tool/BUILD.gn:1]: Please provide a PR description explaining why these changes are made.

"src/app/commands" in particular is not really a path that makes sense.  Having "commands" under "test suites", ok: those are commands you can issue to a test tool.  But "commands" for "src/app" means something completely different (IM commands), normally....  But it's hard to say what the right naming here is without understanding what the goal is.

Comment (jlatusek) [reply examples/chip-tool/BUILD.gn:1]: I made these changes back; I thought that using files from tests in chip-tool was counterintuitive at the beginning. I've limited these changes to only one related to removing dependency between chip-tool and nlunit-test

File: src/app/tests/suites/commands/interaction_model/InteractionModel.h [hunk 1]
```diff
@@ -258,7 +258,7 @@ class InteractionModelCommands
 
             if (mRepeatDelayInMs.HasValue())
             {
-                chip::test_utils::SleepMillis(mRepeatDelayInMs.Value());
```
Comment (tcarmelveilleux) [inline src/app/tests/suites/commands/interaction_model/InteractionModel.h:261]: Please leave `test_utils::` namespace.

We don't want to start seeing standard non-test-code call things like "SleepMillis" here and there.

Comment (jlatusek) [reply src/app/tests/suites/commands/interaction_model/InteractionModel.h]: Ok, I've reverted my changes and left only one related to breaking dependencies.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by arkq — APPROVED** [review]

**Pull request merged**